tests_and_analysis/test/euphonic_test/test_calculate_structure_factor.py

```diff
@@ -17,109 +17,110 @@
     check_structure_factor, get_sf_dir)
 
 
-@pytest.mark.integration
+def get_quartz_fc():
+    return ForceConstants.from_castep(
+        get_castep_path('quartz', 'quartz.castep_bin'))
+
+
+def get_si2_fc():
+    return ForceConstants.from_castep(
+        get_castep_path('Si2-sc-skew', 'Si2-sc-skew.castep_bin'))
+
+
 class TestCalculateStructureFactorFromForceConstants:
 
-    def get_multithreaded_kwargs(self, n_threads):
-        kwargs = {}
-        if n_threads == 0:
-            kwargs['use_c'] = False
-        else:
-            kwargs['use_c'] = True
-            kwargs['n_threads'] = n_threads
-        return kwargs
-
-    @pytest.fixture(params=[0, 1, 2])
-    def get_quartz_qpt_ph_modes_recip(self, request):
-        fc = ForceConstants.from_castep(
-            get_castep_path('quartz', 'quartz.castep_bin'))
-        kwargs = self.get_multithreaded_kwargs(request.param)
-        return fc.calculate_qpoint_phonon_modes(
-            get_test_qpts('split'), asr='reciprocal', **kwargs)
-
-    @pytest.fixture(params=[0, 1, 2])
-    def get_quartz_qpt_ph_modes(self, request):
-        fc = ForceConstants.from_castep(
-            get_castep_path('quartz', 'quartz.castep_bin'))
-        kwargs = self.get_multithreaded_kwargs(request.param)
-        return fc.calculate_qpoint_phonon_modes(
-            get_test_qpts('split'), **kwargs)
-
-    @pytest.fixture(params=[0, 1, 2])
-    def get_si2_qpt_ph_modes(self, request):
-        fc = ForceConstants.from_castep(
-            get_castep_path('Si2-sc-skew', 'Si2-sc-skew.castep_bin'))
-        kwargs = self.get_multithreaded_kwargs(request.param)
-        return fc.calculate_qpoint_phonon_modes(get_test_qpts(), **kwargs)
-
-    @pytest.fixture(params=[0, 1, 2])
-    def get_cahgo2_qpt_ph_modes(self, request):
-        fc = ForceConstants.from_phonopy(
-            path=get_phonopy_path('CaHgO2', ''),
-            summary_name='mp-7041-20180417.yaml')
-        kwargs = self.get_multithreaded_kwargs(request.param)
-        return fc.calculate_qpoint_phonon_modes(get_test_qpts(), **kwargs)
+    tol_kwargs = {'sf_atol': 3e-3,
```

Comment:
Why these tolerances? Magic numbers usually need comment.

Reply:
There is a comment, but stupidly its where these values are used rather than where they're defined. I've moved things around.